Update subsetting code & initial terraform configuration

program-- opened the pull request:
This PR updates the subsetting code by vendoring in the hydrofabric repository and creating a symlink to it rather than manual vendoring.

Additionally, this PR adds an initial terraform configuration to resolve #6.

@ZacharyWills please review the TF code in `terraform/` if you get a chance, in case there is anything I'm missing and/or need to add or refactor 😄

ZacharyWills commented on terraform/main.tf:
```diff
@@ -0,0 +1,252 @@
+}
+
+data "aws_vpc" "default_vpc" {
+  default = true
```
Only worry here is the co-location using the default VPC. Ideally we would have a specific VPC for the Lynker-spatial products and services. Not an immediate need but certainly an eventual one.

program-- replied:
Sounds good, should be an easy change! 😄 Though I do have a few questions on resource "ownership" with terraform, so maybe I'll reach out to talk to you about that -- specifically, how should resource creation/state be maintained for "site"-wide resources, like a VPC? (i.e. site in this case being lynker-spatial)

ZacharyWills commented on terraform/main.tf:
I got a few misconfiguration items (it'll work, but these are probably good things to add before we push it up). 
<img width="1613" alt="Screenshot 2023-10-31 at 19 38 06" src="https://github.com/LynkerIntel/hfsubset/assets/60660437/74ec4a47-1793-47b1-a3d9-58b2711b6fd6">
Would recommend installing Trivy (https://github.com/aquasecurity/trivy) which is how I got the above screenshot and knocking out some of the low hanging fruit, and I'll contribute to that as well as best I can find time to.
If I do find some time later this week I can work on installing Trivy for the repo using a GitHub action (https://github.com/aquasecurity/trivy-action)
The big benefit of Trivy for security being that it scans the config without needing to have it deployed or running.

On a cost note, this is well worth the money (projection generated with Infracost):
<img width="786" alt="Screenshot 2023-10-31 at 19 43 01" src="https://github.com/LynkerIntel/hfsubset/assets/60660437/acf28b53-8fe6-44f2-8fd1-f87f44f7525f">

ZacharyWills replied:
I'll finish my review in the morning. I would say this is good to deploy in it's current state with some cleanup needed with regards to logging, documentation (which is most of the Trivy items) and an eventual move to a bespoke VPC instead of using the default.

program-- commented:
@ZacharyWills I've updated the terraform to include a new VPC. Also, fixed some configuration issues with ECS -- the pricing went up to ~$55/month because I had set 4 CPU units instead of a full vCPU 🤦🏾 changed it now to be a full vCPU, which caused the price increase. If you could do another quick review, that'd be great!

mikejohnson51 reviewed (commented):
Looks good to me! Thanks @program--

ZacharyWills commented:
Terraform looked good, just forgot to submit my comment. 😄
